docs/build.md

```diff
@@ -0,0 +1,226 @@
+# Build
+
+- [Overview](#overview)
+- [Build Controller](#build-controller)
+- [Configuring a Build](#configuring-a-build)
+  - [Defining the Source](#defining-the-source)
+  - [Defining the Strategy](#defining-the-strategy)
+  - [Defining the Builder or Dockerfile](#defining-the-builder-or-dockerfile)
+  - [Defining Resources](#defining-resources)
+  - [Defining the Output](#defining-the-output)
+
+## Overview
+
+A `Build` resource allows the user to define:
+
+- source
+- strategy
+- builder
+- dockerfile
+- resources
+- output
+
+A `Build` is available within a namespace.
+
+## Build Controller
+
+The controller watches for:
+
+- Updates on the `Build` resource (_CRD instance_)
+
+When the controller reconciles it:
+
+- Validates if the referenced `StrategyRef` exists.
+- Validates if the container `registry` output secret exists.
+
+## Configuring a Build
+
+The `Build` definition supports the following fields:
+
+- Required:
+  - [`apiVersion`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the API version, for example `build.dev/v1alpha1`.
+  - [`kind`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the Kind type, for example `Build`.
+  - [`metadata`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Metadata that identify the CRD instance, for example the name of the `Build`.
+  - `spec.source.URL` - Refers to the Git repository containing the source code.
+  - `spec.strategy` - Refers to the `BuildStrategy` to be used, see the [examples](../samples/buildstrategy)
+  - `spec.builder` - Refers to the image containing the build tools. (_when used, `spec.dockerfile` should not be used_)
+  - `spec.dockerfile` - Path to a Dockerfile to be used for building an image. (_when used, `spec.builder` should not be used_)
```

Comment:
I am not sure if this is accurate. So, it is `spec.builder.image` and feel like this is relevant for non-Docker builds (buildpack). The `spec.dockerfile` property I so far only associate with Docker base builds (kaniko and buildah) to specify the path relative to `source.contextDir` if the Dockerfile is not in the contextDir and/or not called Dockerfile. And it is optional = if there is a Dockerfile in the contextDir, it can be omitted.

Reply:
yes, I think I understand what you mean, I will move this into something like:
```
  - `spec.builder.image` - Refers to the image containing the build tools to build the source code. (_Use this path for Dockerless strategies_)
  - `spec.dockerfile` - Path to a Dockerfile to be used for building an image. (_Use this path for strategies that require a Dockerfile_)
```
and I added a new optional one:
```
  - `spec.source.contextDir` - Refers to the location of the `Dockerfile`. It can be used if `spec.dockerfile` is specified.
```

Reply:
Looks better now. You can move `spec.dockerfile` to the optional fields section. Its default value is `Dockerfile`. I guess that in > 90 % of the cases, a user will not specify it because the Dockerfile is called `Dockerfile` and is in the `contextDir`.

Reply:
good point, done